src/sentry/integrations/client.py

```diff
@@ -236,3 +237,19 @@ def _request(self, method, path, **kwargs):
         self.auth.refresh_token()
         kwargs = self.bind_auth(**kwargs)
         return ApiClient._request(self, method, path, **kwargs)
+
+
+class OAuth2ApiClient(ApiClient):
+    def __init__(self, identity, *args, **kwargs):
+        super(OAuth2ApiClient, self).__init__(*args, **kwargs)
+        self.identity = identity
```

Comment:
I wonder if maybe we could actually make this more of a mixin `OAuth2RefreshingMixin`, so instead of this extending ApiClient, it's just a mixin that has this functionality. Since this also ties the client specifically to using identities.